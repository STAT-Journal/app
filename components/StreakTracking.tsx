// components/Entry/StreakTracker.tsx
import React from "react";
import { View } from "react-native";
import { Card, Title } from "react-native-paper";
import { StreakTrackerStyles } from "@/styles/styles";

interface StreakTrackerProps {
  streak: number;
}

const StreakTracker: React.FC<StreakTrackerProps> = ({ streak }) => {
  return (
    <View style={StreakTrackerStyles.container}>
      <Card>
        <Card.Content style={StreakTrackerStyles.cardContent}>
          <Title>ðŸŽ‰{streak}</Title>
        </Card.Content>
      </Card>
    </View>
  );
};

export default StreakTracker;
