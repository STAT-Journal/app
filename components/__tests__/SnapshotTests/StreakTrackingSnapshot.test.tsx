import React from 'react';
import renderer from 'react-test-renderer'; // Import the 'renderer' object from the 'react-test-renderer' package

import StreakTracking from '../../StreakTracking';

describe('<StreakTracking />', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<StreakTracking streak={32}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});