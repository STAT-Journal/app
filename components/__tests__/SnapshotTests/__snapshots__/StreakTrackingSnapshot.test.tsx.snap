// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StreakTracking /> renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "justifyContent": "center",
      "padding": 16,
    }
  }
>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "rgba(247, 243, 249, 1)",
        "borderRadius": 12,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.15,
        "shadowRadius": 3,
      }
    }
    testID="card-container-outer-layer"
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "rgba(247, 243, 249, 1)",
          "borderRadius": 12,
          "elevation": 1,
          "flex": undefined,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 1,
        }
      }
      testID="card-container"
    >
      <View
        style={
          [
            {
              "flexShrink": 1,
            },
            undefined,
          ]
        }
        testID="card"
      >
        <View
          borderRadiusStyles={{}}
          style={
            [
              {
                "paddingHorizontal": 16,
              },
              {
                "paddingVertical": 16,
              },
              {
                "alignItems": "center",
                "color": "black",
                "flexDirection": "row",
              },
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "rgba(28, 27, 31, 1)",
                },
                {
                  "textAlign": "left",
                },
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 0.87)",
                    "writingDirection": "ltr",
                  },
                  [
                    {
                      "fontSize": 20,
                      "letterSpacing": 0.15,
                      "lineHeight": 30,
                      "marginVertical": 2,
                    },
                    undefined,
                  ],
                ],
              ]
            }
          >
            ðŸŽ‰
            32
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
